From b16536e13de08bfb2124bf56d56dc9b8ef12c1c5 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Thu, 16 Sep 2021 06:15:14 +0300
Subject: [PATCH] [SW-56221] gaudi2: arc: ts shutdown fix

Currently for gaudi2 onix platform, ts init is a dummy function
but ts shutdown is an actual function. This mismatch leads to
ts shutdown to access and modify illegal address which leads
to an exception.

This patch aligns init/shutdown api implementation for onix.

Change-Id: Ib7f8da2c821db05d473cb21d2f5771e9d3a95f87
---

diff --git a/arch/gaudi2/common/onix.c b/arch/gaudi2/common/onix.c
index 2e0cdf2..f6289e6 100644
--- a/arch/gaudi2/common/onix.c
+++ b/arch/gaudi2/common/onix.c
@@ -42,3 +42,5 @@
 }
 
 void hl_dev_ts_init(void) {}
+
+void hl_ts_shutdown(void) {}
diff --git a/lib/preboot/thermal.c b/lib/preboot/thermal.c
index 296f8ed..81160a8 100644
--- a/lib/preboot/thermal.c
+++ b/lib/preboot/thermal.c
@@ -253,7 +253,7 @@
 __weak void hl_ts_pvt_shutdown(int ts) {}
 __weak void hl_ts_reset(void) {}
 
-void hl_ts_shutdown(void)
+__weak void hl_ts_shutdown(void)
 {
 	u32 ts;
 
